app/main.py

```diff
@@ -1,3 +1,57 @@
+import json
+
+from app.customers import Customer
+from app.shops import Shop
+
+
+FILE_NAME = "app/config.json"
+
+
 def shop_trip():
-    # write your code here
-    pass
+    shops, customers = create_shops_and_customers(FILE_NAME)
+
+    for customer in customers:
+        chosen_shop, trip_costs = get_customer_shop(customer, shops)
+        if chosen_shop is not None:
+            print(f"{customer.name} rides to {chosen_shop}\n")
+            chosen_shop.sell_products(customer)
+            print(f"{customer.name} rides home")
+            left_money = customer.pay_for_trip(trip_costs)
+            print(f'{customer.name} now has {left_money} dollars\n')
+        else:
+            print(f"{customer.name} doesn't have enough "
+                  f"money to make purchase in any shop")
+
+
+def get_customer_shop(customer: Customer, shops: list):
+    print(f"{customer.name} has {customer.money} dollars")
+
+    shops_cost = {}
+    for shop in shops:
+        costs = customer.trip_costs(shop)
+        print(f"{customer.name}'s trip to the {shop.name} costs {costs}")
+        shops_cost[costs] = shop
+    shops_cost = dict(sorted(shops_cost.items()))
+
+    for costs, shop in shops_cost.items():
+        if costs <= customer.money:
+            return shop, costs
+    return None, None
```

Comment:
maybe its better to create method for Customer.
If you create method for trip in Customer, why choose shop not in class too?